### ISSUE TITLE
Don't warn in react/jsx-props-no-spreading when all properties are listed explicitly.

### ISSUE THREAD

**Issue by MattSmiglarski** [yannickcr/eslint-plugin-react#2439]
A syntactic workaround for avoiding repetition in the following JSX:

```
<Foo prop1={prop1} prop2={prop2} prop3={prop3} />
```

..is to instead write:
```
<Foo {... { prop1, props2, prop3 }} />
```

The reason for react/jsx-props-no-spreading is to encourage being explicit with properties, while allowing exceptions to be configured for Higher Ordered Components. The above example is not a HOC, but is explicit about the properties.

Therefore, there should be a configuration to allow this case ("spreading of an inline object literal that does not itself contain any spreads"), perhaps even as a default.

That said, ideally JSX would provide a nicer non-repetitive syntax for the above.

**Comment by ljharb** [comment]
An option to allow this case sounds great, and does not conflict with the impetus for the rule - that implicit spreading is an antipattern.

**Comment by pawelnvk** [comment]
I add PR, that might help here. I thought for a while about allowing also to handle cases when some given props are assigned to dedicated variable and than spread, e.g.:
```jsx
const modalProps = { prop1, prop2, prop3 };

return <Modal {...modalProps } />
```
I feel that this could be helpful, yet current implementation is good enough to solve your problem without extending implementation complexity by far. What do you think guys?

**Comment by ljharb** [comment]
@richardson-trevor there's no good way to pick up on that, and it's debatable whether that's a good pattern anyways. For that case, use an eslint override comment.

**Comment by bzamora** [comment]
> ```js-jsx
> const modalProps = { prop1, prop2, prop3 };
> 
> return <Modal {...modalProps } />
> ```

Allowing this would be really useful

**Comment by ljharb** [comment]
@bzamora it's done in #2449.

**Comment by bzamora** [comment]
@ljharb reading upstream, it seems scenario of using a dedicated variable and then spread isn't covered? If it is, then 👍 

See https://github.com/yannickcr/eslint-plugin-react/issues/2439#issuecomment-538664548

**Comment by ljharb** [comment]
No, it's not, but that would only even be possible to handle if the variable is defined in the same file.

I'd be open to a PR exploring that, but this issue should stay closed, as it's resolved.

**Comment by ljharb** [comment]
@arathjz yes, see #2449.

**Comment by arathjz** [comment]
@ljharb #2449 does not cover 

```
const modalProps = { prop1, prop2, prop3 };

return <Modal {...modalProps } />
````

it only covers explicit props as 

```
return <Modal {...{ prop1, prop2, prop3 }} />
```

According to #2249  tests that case is still displaying `Prop spreading is forbidden`

![image](https://user-images.githubusercontent.com/20747535/78705430-12f99300-78d3-11ea-9b68-88375ee1b1cc.png)

**Comment by arathjz** [comment]
@ljharb I'll be opening a pull request to see if I can propose this as part of `explicitSpread`
